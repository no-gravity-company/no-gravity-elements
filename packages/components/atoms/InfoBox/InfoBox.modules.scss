:host {
    background-color: blue;

    .main {
        background-color: aquamarine;
    }
}
