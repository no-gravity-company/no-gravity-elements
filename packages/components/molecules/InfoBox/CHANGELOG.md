# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [5.0.10](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/info-box@5.0.7...@no-gravity-elements/info-box@5.0.10) (2024-03-30)

### Bug Fixes

- fix cicd publish ([10b9043](https://github.com/no-gravity-company/no-gravity-elements/commit/10b9043521df21266dda1d07c7c7cb60606997d4))

## [5.0.9](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/info-box@5.0.7...@no-gravity-elements/info-box@5.0.9) (2024-03-30)

### Bug Fixes

- fix cicd publish ([f084cef](https://github.com/no-gravity-company/no-gravity-elements/commit/f084cefcecb4f411e11da6413aa4fa9f6fbdda72))

## [5.0.7](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/info-box@5.0.6...@no-gravity-elements/info-box@5.0.7) (2024-03-30)

**Note:** Version bump only for package @no-gravity-elements/info-box

## [5.0.6](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/info-box@5.0.1...@no-gravity-elements/info-box@5.0.6) (2024-03-29)

**Note:** Version bump only for package @no-gravity-elements/info-box

## [5.0.1](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/info-box@5.0.0...@no-gravity-elements/info-box@5.0.1) (2024-03-11)

### Bug Fixes

- Ignore changelogs in lerna config ([9f9c7ee](https://github.com/no-gravity-company/no-gravity-elements/commit/9f9c7ee07e4e05f8dfe9c934bf884515ee8d0732))
