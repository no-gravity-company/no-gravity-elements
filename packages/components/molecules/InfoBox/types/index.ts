export interface InfoBoxProps {
  label: string;
  testProp?: string;
  coolProp?: number;
  isInformative?: boolean;
}
