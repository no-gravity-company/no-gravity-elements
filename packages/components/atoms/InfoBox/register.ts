import register from 'preact-custom-element';
import InfoBox from './InfoBox';

const alreadyDefined = (tagName: string) =>
    customElements.get(tagName) !== undefined;
if (!alreadyDefined('nge-info-box')) {
    register(InfoBox, 'nge-info-box', [], { shadow: true });
}
