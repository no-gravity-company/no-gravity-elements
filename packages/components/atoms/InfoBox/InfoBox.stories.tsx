import { Story } from '@storybook/preact';
import { h } from 'preact';
import { action } from '@storybook/addon-actions';
import './lib/index';
import { InfoBoxProps } from './types';

export default {
  title: 'InfoBox',
  // decorators: [withCustomEventListeners],
  parameters: {
    // Specify that the component is a Web Component
    webComponents: {
      // Set the tag name of the Web Component
      tagName: 'nge-info-box',
    },
  },
  controls: {},
};

const Template: Story<InfoBoxProps> = (props) => {
  return <nge-info-box {...props} onButtonClick={() => action('onButtonClick')()} />;
};

export const Default = Template;
Default.args = {
  label: 'Default label',
};
