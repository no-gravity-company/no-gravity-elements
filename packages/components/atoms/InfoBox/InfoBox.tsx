import { signal } from '@preact/signals';
import { Fragment, FunctionComponent, h } from 'preact';
import { InfoBoxProps } from './types';

const inputValue = signal('');

const InfoBox: FunctionComponent<InfoBoxProps> = ({ label }: InfoBoxProps) => {
    const handleInputChange = (event: Event) => {
        const inputElement = event.target as HTMLInputElement;
        inputValue.value = inputElement.value;
    };

    const handleDivClick = () => {
        const event = new CustomEvent('buttonClick', {
            detail: 'Event detail',
        });
        window.dispatchEvent(event);
    };

    return (
        <div>
            <span>HEY</span>
            <span>HO</span>
            <span>{label}</span>
            <input type="text" value={inputValue} onInput={handleInputChange} />
            <span>{inputValue}</span>
            <button onClick={handleDivClick}></button>
        </div>
    );
};

export default InfoBox;
