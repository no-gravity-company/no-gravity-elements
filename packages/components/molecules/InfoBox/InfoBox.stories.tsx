import { h } from 'preact';
import { Story } from '@storybook/preact';
import { html } from 'lit-html';

import { InfoBoxProps } from '@molecules/InfoBox/types';

import '@no-gravity-elements/info-box';

export default {
  title: 'molecules/InfoBox',
  // decorators: [withCustomEventListeners],
  component: 'nge-info-box',
  parameters: {
    // Specify that the component is a Web Component
    webComponents: {
      // Set the tag name of the Web Component
      tagName: 'nge-info-box',
    },
    actions: {
      handles: ['ButtonClick'],
    },
  },
  controls: {},
};
export const Default: Story<InfoBoxProps> = (args): any => {
  return html` <nge-info-box /> `;
};

Default.args = {
  label: 'Default label',
};
