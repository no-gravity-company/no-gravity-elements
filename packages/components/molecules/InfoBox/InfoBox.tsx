import { Fragment, FunctionComponent, h } from 'preact';
import { signal } from '@preact/signals';

import { InfoBoxProps } from '@molecules/InfoBox/types';

const inputValue = signal('');

/**
 * @element nge-info-box
 * @slot - Default slot content.
 * @slot header - Header slot content.
 * @cssprop --primary-color - The primary color of the element.
 *
 * @fires my-event - Fired when the element is clicked.
 *
 * @prop {string} [label] - The label of the element.
 *
 * @example
 * <nge-info-box label="Hello world">
 *   <div slot="header">Header content</div>
 *   <div>Default content</div>
 * </nge-info-box>
 */
const InfoBox: FunctionComponent<InfoBoxProps> = ({ label }: InfoBoxProps) => {
  const handleInputChange = (event: Event) => {
    const inputElement = event.target as HTMLInputElement;
    inputValue.value = inputElement.value;
  };

  const handleDivClick = (event: Event) => {
    // TODO extract to function in tools repo -> publish(eventName, payload, element)
    if (event && event.target) {
      const customEvent = new CustomEvent('ButtonClick', {
        detail: {
          'nge-event': true,
        },
        // TODO try to consume it without those attrs (maybe they are only needed if we want to cacth them with a listener)
        bubbles: true,
        composed: true,
      });
      event.target.dispatchEvent(customEvent);
    }
  };

  return (
    <Fragment>
      <span className='main'>HEY</span>
      <span>{label}</span>
      <input type='text' value={inputValue} onInput={handleInputChange} />
      <span>{inputValue}</span>
      <button onClick={handleDivClick} />
    </Fragment>
  );
};

export default InfoBox;
