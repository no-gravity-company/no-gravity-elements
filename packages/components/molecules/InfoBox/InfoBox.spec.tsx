import { h } from 'preact';
import { shallow } from 'enzyme';

import InfoBox from '@molecules/InfoBox/InfoBox';

describe('InfoBox', () => {
  it.each`
    label
    ${'foo'}
    ${'bar'}
  `('should match the snapshot', ({ label }) => {
    const wrapper = shallow(<InfoBox label={label} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with the provided label', () => {
    const label = 'Test Label';
    const wrapper = shallow(<InfoBox label={label} />);
    expect(wrapper.find('span').at(0).text()).toEqual(label);
  });
});
