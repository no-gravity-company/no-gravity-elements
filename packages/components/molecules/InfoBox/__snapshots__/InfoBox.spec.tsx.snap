// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox should match the snapshot 1`] = `
[
  <span
    className="main"
  >
    HEY
  </span>,
  <span>
    foo
  </span>,
  <input
    onInput={[Function]}
    type="text"
    value=""
  />,
  <span>
    <_st
      data={
        {
          "i": 0,
          "n": undefined,
          "t": {
            "S": [Circular],
            "e": undefined,
            "i": 0,
            "n": undefined,
            "p": {
              "S": {
                "i": 0,
                "n": undefined,
                "t": [Circular],
                "v": [Circular],
              },
              "e": undefined,
              "i": 0,
              "n": [Circular],
              "p": undefined,
              "r": undefined,
              "t": b {
                "f": 32,
                "o": undefined,
                "s": [Circular],
                "u": undefined,
                "x": [Function],
              },
              "x": undefined,
            },
            "r": undefined,
            "t": b {
              "f": 32,
              "o": undefined,
              "s": {
                "S": {
                  "i": 0,
                  "n": undefined,
                  "t": [Circular],
                  "v": [Circular],
                },
                "e": undefined,
                "i": 0,
                "n": [Circular],
                "p": undefined,
                "r": undefined,
                "t": [Circular],
                "x": undefined,
              },
              "u": undefined,
              "x": [Function],
            },
            "x": undefined,
          },
          "v": "",
        }
      }
    />
  </span>,
  <button
    onClick={[Function]}
  />,
]
`;

exports[`InfoBox should match the snapshot 2`] = `
[
  <span
    className="main"
  >
    HEY
  </span>,
  <span>
    bar
  </span>,
  <input
    onInput={[Function]}
    type="text"
    value=""
  />,
  <span>
    <_st
      data={
        {
          "i": 0,
          "n": undefined,
          "t": {
            "S": [Circular],
            "e": undefined,
            "i": 0,
            "n": undefined,
            "p": {
              "S": {
                "i": 0,
                "n": undefined,
                "t": [Circular],
                "v": [Circular],
              },
              "e": undefined,
              "i": 0,
              "n": [Circular],
              "p": undefined,
              "r": undefined,
              "t": b {
                "f": 32,
                "o": undefined,
                "s": [Circular],
                "u": undefined,
                "x": [Function],
              },
              "x": undefined,
            },
            "r": undefined,
            "t": b {
              "f": 32,
              "o": undefined,
              "s": {
                "S": {
                  "i": 0,
                  "n": undefined,
                  "t": [Circular],
                  "v": [Circular],
                },
                "e": undefined,
                "i": 0,
                "n": [Circular],
                "p": undefined,
                "r": undefined,
                "t": [Circular],
                "x": undefined,
              },
              "u": undefined,
              "x": [Function],
            },
            "x": {
              "S": [Circular],
              "e": [Circular],
              "i": 0,
              "n": undefined,
              "p": {
                "S": {
                  "i": 0,
                  "n": undefined,
                  "t": [Circular],
                  "v": [Circular],
                },
                "e": undefined,
                "i": 0,
                "n": [Circular],
                "p": undefined,
                "r": undefined,
                "t": b {
                  "f": 32,
                  "o": undefined,
                  "s": [Circular],
                  "u": undefined,
                  "x": [Function],
                },
                "x": undefined,
              },
              "r": undefined,
              "t": b {
                "f": 32,
                "o": undefined,
                "s": {
                  "S": {
                    "i": 0,
                    "n": undefined,
                    "t": [Circular],
                    "v": [Circular],
                  },
                  "e": undefined,
                  "i": 0,
                  "n": [Circular],
                  "p": undefined,
                  "r": undefined,
                  "t": [Circular],
                  "x": undefined,
                },
                "u": undefined,
                "x": [Function],
              },
              "x": undefined,
            },
          },
          "v": "",
        }
      }
    />
  </span>,
  <button
    onClick={[Function]}
  />,
]
`;
